src/crd.cr

```diff
@@ -148,8 +153,10 @@ module Kubernetes
                       raise "Array type specification for #{name.inspect} must contain an `items` key"
                     end
                   when "object"
-                    if preserve_unknown_fields?
+                    if preserve_unknown_fields? || additional_properties == true
                       "Hash(String, JSON::Any)"
+                    elsif apspec = additional_properties.as?(Spec)
+                      "Hash(String, #{apspec.to_crystal(name)})"
```

Comment:
This seems _exactly_ like what I had in mind. 💯 I'll test it a bit and if it seems right, I'll go ahead and merge.